### PR TITLE
fix dolt log pager panics with ctrl+c on Windows

### PR DESCRIPTION
**Pull request by jennifersp**
added catching Interrupts for `dolt log` as it panics with Ctrl+C on Windows with `more` command

### CODE REVIEW

**Review by zachmu — COMMENTED** [review]
Write a more descriptive PR subject and details. What problem does this solve?

**Review by VinaiRachakonda — APPROVED** [review]

**Comment by VinaiRachakonda** [comment]
Need tests

File: go/libraries/doltcore/sqle/tables.go [hunk 1]
```diff
@@ -1338,6 +1338,7 @@ func (t *AlterableDoltTable) CreateForeignKey(
 	if err != nil {
 		return err
 	}
+	// TODO : ALTER TABLE statements should return error for ForeignKeyConstraints
```
Comment (VinaiRachakonda) [inline go/libraries/doltcore/sqle/tables.go:1341]: Shouldn't this be in the skipped bats test pr?

File: go/store/util/outputpager/page_output.go [hunk 2]
```diff
@@ -75,9 +83,28 @@ func Start() *Pager {
...
+			default:
+			}
+		}
+	}()
+
 	go func() {
 		err := cmd.Wait()
-		d.Chk.NoError(err)
+		if err != nil {
+			fmt.Println("cmd.Wait: ", err)
```
Comment (VinaiRachakonda) [inline go/store/util/outputpager/page_output.go:106]: Better error message.

File: go/store/util/outputpager/page_output.go [hunk 3]
```diff
@@ -110,3 +137,7 @@ func RegisterOutputpagerFlags(flags *flag.FlagSet) {
 func IsStdoutTty() bool {
 	return goisatty.IsTerminal(os.Stdout.Fd())
 }
+
+func SetTestingArg(s bool) {
```
Comment (VinaiRachakonda) [inline go/store/util/outputpager/page_output.go:141]: Doesn't need to be exported

File: go/cmd/dolt/commands/log_test.go [hunk 4]
```diff
@@ -38,3 +44,47 @@ func TestLog(t *testing.T) {
...
+		formattedDesc := "\n\n\t" + strings.Replace(cMeta.Description, "\n", "\n\t", -1) + "\n\n"
+		pager.Writer.Write([]byte(fmt.Sprintf(formattedDesc)))
+	}
+
+	process, err := os.FindProcess(syscall.Getpid())
+	require.NoError(t, err)
+
+	err = process.Signal(syscall.SIGTERM)
+	require.NoError(t, err)
+
```
Comment (VinaiRachakonda) [inline go/cmd/dolt/commands/log_test.go:89]: Remove this empty line

**Review by VinaiRachakonda — APPROVED** [review]
LGTM! Thanks for toughing this out.

File: go/store/util/outputpager/page_output.go (hunk 3 above)
Comment (jennifersp) [reply go/store/util/outputpager/page_output.go:141]: it's used in `TestLogSigTerm` in log_test.go

**Pull request merged**